src/basic/process-util.c

```diff
@@ -922,6 +922,64 @@ int ioprio_parse_priority(const char *s, int *ret) {
         return 0;
 }
 
+/* The cached PID, possible values:
+ *
+ *     == 0          → cache not initialized yet
+ *     == (pid_t) -1 → some thread is initializing it at the moment
+ *     any other     → the cached PID
+ */
+static pid_t cached_pid = 0;
```

Comment:
Of course these assignments of 0 should just use the appropriate define if added...